### PR TITLE
PV EventHandler should handle the deletion of PV

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
PV EventHandler should handle the deletion of PV by removing finalizer.

```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @msau42

**Comment by tedyu** [comment]
@msau42
Please review

File: pkg/controller/volume/pvprotection/pv_protection_controller.go [hunk 1]
```diff
@@ -69,11 +69,37 @@ func NewPVProtectionController(pvInformer coreinformers.PersistentVolumeInformer
 		UpdateFunc: func(old, new interface{}) {
 			e.pvAddedUpdated(new)
 		},
+		DeleteFunc: func(obj interface{}) {
```
Comment (msau42) [inline pkg/controller/volume/pvprotection/pv_protection_controller.go]: I'm a little confused.  If PV object is deleted, then there is no more PV object to remove the finalizer on?

Comment (tedyu) [reply pkg/controller/volume/pvprotection/pv_protection_controller.go]: Please look at line 83 below - we obtain the PV from tombstone.

Comment (msau42) [reply pkg/controller/volume/pvprotection/pv_protection_controller.go]: But I'm not sure if it's actually possible to update an object (that removes the finalizer) that has been deleted already.

Comment (tedyu) [reply pkg/controller/volume/pvprotection/pv_protection_controller.go]: @msau42 
Looking at Controller#removeFinalizer, removing finalizer from pvClone.ObjectMeta.Finalizers is feasible in this case.

Comment (tedyu) [reply pkg/controller/volume/pvprotection/pv_protection_controller.go]: @msau42 
Please take a look at pkg/controller/service/service_controller.go , around line 310.

**Pull request closed without merging**